queries/python/indents.scm

```diff
@@ -18,67 +14,94 @@
-] @indent
-
-((for_statement) @indent
- (#set! "immediate_indent" 1))
-((if_statement) @indent
- (#set! "immediate_indent" 1))
-((while_statement) @indent
- (#set! "immediate_indent" 1))
-((try_statement) @indent
- (#set! "immediate_indent" 1))
-((ERROR "try" ":") @indent
- (#set! "immediate_indent" 1))
-((function_definition) @indent
- (#set! "immediate_indent" 1))
-((class_definition) @indent
- (#set! "immediate_indent" 1))
-((with_statement) @indent
- (#set! "immediate_indent" 1))
+] @indent.begin
+
+((list) @indent.align
+ (#set! indent.open_delimiter "[")
+ (#set! indent.close_delimiter "]")
+)
+((dictionary) @indent.align
+ (#set! indent.open_delimiter "{")
+ (#set! indent.close_delimiter "}")
+)
+((set) @indent.align
+ (#set! indent.open_delimiter "{")
+ (#set! indent.close_delimiter "}")
+)
+
+((for_statement) @indent.begin
+ (#set! indent.immediate 1))
+((if_statement) @indent.begin
+ (#set! indent.immediate 1))
+((while_statement) @indent.begin
+ (#set! indent.immediate 1))
+((try_statement) @indent.begin
+ (#set! indent.immediate 1))
+(ERROR "try" ":" @indent.begin (#set! indent.immediate 1))
+((function_definition) @indent.begin
+ (#set! indent.immediate 1))
+((class_definition) @indent.begin
+ (#set! indent.immediate 1))
+((with_statement) @indent.begin
+ (#set! indent.immediate 1))
 
 (if_statement
-  condition: (parenthesized_expression) @aligned_indent
-  (#set! "delimiter" "()")
-  (#set! "final_line_indent" 1) ; parenthesized_expression already indented
+  condition: (parenthesized_expression) @indent.align
+  (#set! indent.open_delimiter "(")
+  (#set! indent.close_delimiter ")")
+  (#set! indent.avoid_last_matching_next 1)
 )
 (while_statement
-  condition: (parenthesized_expression) @aligned_indent
-  (#set! "delimiter" "()")
-  (#set! "final_line_indent" 1) ; parenthesized_expression already indented
+  condition: (parenthesized_expression) @indent.align
+  (#set! indent.open_delimiter "(")
+  (#set! indent.close_delimiter ")")
+  (#set! indent.avoid_last_matching_next 1)
 )
 
-((ERROR "(" . (_)) @aligned_indent
- (#set! "delimiter" "()"))
-((argument_list ")" @indent_end) @aligned_indent
- (#set! "delimiter" "()"))
-((parameters) @aligned_indent
- (#set! "delimiter" "()")
- (#set! "final_line_indent" 1))
-((tuple ")" @indent_end) @aligned_indent
- (#set! "delimiter" "()"))
+(ERROR "(" @indent.align (#set! indent.open_delimiter "(") (#set! indent.close_delimiter ")") . (_)) 
+((argument_list) @indent.align
+ (#set! indent.open_delimiter "(")
+ (#set! indent.close_delimiter ")"))
+((parameters) @indent.align
+ (#set! indent.open_delimiter "(")
+ (#set! indent.close_delimiter ")")
+ (#set! indent.avoid_last_matching_next 1))
+((tuple) @indent.align
+ (#set! indent.open_delimiter "(")
+ (#set! indent.close_delimiter ")"))
 
-(list "]" @indent_end)
-(dictionary "}" @indent_end)
-(set "}" @indent_end)
+(ERROR "[" @indent.align (#set! indent.open_delimiter "[") (#set! indent.close_delimiter "]") . (_)) 
+(list "]" @indent.end)
```

Comment:
This should not be needed if you are using `@indent.align`, or has something changed again?

Reply:
Yes, you are right.  Thanks